pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,267 @@
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+	"github.com/openshift/origin/pkg/util/parallel"
+)
+
+// ChainDescriber generates extended information about a chain of
+// dependencies of an image stream
+type ChainDescriber struct {
+	client.Interface
+	Output string
+}
+
+// NewChainDescriber returns a new ChainDescriber
+func NewChainDescriber(c client.Interface, out string) *ChainDescriber {
+	return &ChainDescriber{Interface: c, Output: out}
+}
+
+// MakeGraph will create the graph of all build configurations and the image streams
+// they point to via image change triggers in the provided namespace
+func (d *ChainDescriber) MakeGraph(namespace string) (osgraph.Graph, error) {
+	g := osgraph.New()
+
+	loaders := []GraphLoader{
+		&bcLoader{namespace: namespace, lister: d.Interface},
+		&isFromBuildConfigLoader{namespace: namespace, lister: d.Interface, bc: &bcLoader{namespace: namespace, lister: d.Interface}},
+	}
+	loadingFuncs := []func() error{}
+	for _, loader := range loaders {
+		loadingFuncs = append(loadingFuncs, loader.Load)
+	}
+
+	if errs := parallel.Run(loadingFuncs...); len(errs) > 0 {
+		return g, utilerrors.NewAggregate(errs)
+	}
+
+	for _, loader := range loaders {
+		loader.AddToGraph(g)
+	}
+
+	buildedges.AddAllInputOutputEdges(g)
+	imageedges.AddAllImageStreamRefEdges(g)
+
+	return g, nil
+}
+
+func (d *ChainDescriber) Describe(namespace, input string) (string, error) {
+	g, err := d.MakeGraph(namespace)
+	if err != nil {
+		return "", err
+	}
+	name, tag, all, err := parseInput(input)
+	if err != nil {
+		return "", err
+	}
+	if all {
+		// TODO: Dump all the graph
+		return "", nil
+	}
+
+	// Find out if the provided stream exists
+	exists := false
+	var stream graph.Node
+	for _, node := range g.NodesByKind(imagegraph.ImageStreamNodeKind) {
+		if g.GraphDescriber.Name(node) == name {
+			stream = node
+			exists = true
+			break
+		}
+	}
+	if !exists {
+		return "", fmt.Errorf("image stream %q does not exist", name)
+	}
+
+	if d.Output == "dot" {
+		return dotDump(g, stream, tag)
+	}
+	return g.String(), nil
+}
+
+// parseInput pasrses user input and returns a stream name, a tag
+// and in case of empty input a true condition meaning the caller
+// should look for all imageStreams in the current namespace
+// TODO: Move this under build-chain validation
+func parseInput(input string) (string, string, bool, error) {
+	args := strings.Split(input, ":")
+	if len(args[0]) == 0 {
+		return "", "", true, nil
+	}
+
+	var name, tag string
+	switch len(args) {
+	case 1:
+		name = args[0]
+		tag = imageapi.DefaultImageTag
+	case 2:
+		name = args[0]
+		tag = args[1]
+	default:
+		return "", "", false, fmt.Errorf("invalid stream:tag combination %q", input)
+	}
+	return name, tag, false, nil
+}
+
+// isFromBuildConfigLoader loads image streams found in image change triggers
+// of all build configurations in the provided namespace
+type isFromBuildConfigLoader struct {
```

Comment:
@deads2k do we even need this? It seems to me that bcs and their outputs are enough...